File: algorithms/interview-preparation-kit-2/php/code-response/spec/Valleys/Valleys.spec.php
<?php

use Valleys\Valleys;

describe('Valleys', function () {
  it('counting valleys (1)', function () {
    expect(Valleys::count('DU'))->toEqual(1);
  });

  it('counting valleys (2)', function () {
    expect(Valleys::count('DUDU'))->toEqual(2);
  });

  it('counting valleys (3)', function () {
    expect(Valleys::count('UUUDU'))->toEqual(0);
  });

  it('counting valleys (4)', function () {
    expect(Valleys::count('UDDDUDUU'))->toEqual(1);
  });

  it('counting valleys (5)', function () {
    expect(Valleys::count('DDUUDDUDUUUD'))->toEqual(2);
  });
});

File: algorithms/interview-preparation-kit-2/ruby/code-response/spec/Valleys/valleys.spec.rb
require_relative "../../src/Valleys/valleys"

describe Valleys do
	describe ".count" do
		context "Count how many valleys exists" do
			it "return 1" do
				expect(Valleys.count("DU")).to eql(1)
			end
		end

		context "Count how many valleys exists" do
			it "return 2" do
				expect(Valleys.count("DUDU")).to eql(2)
			end
		end
		
		context "Count how many valleys exists" do
			it "return 0" do
				expect(Valleys.count("UUUDU")).to eql(0)
			end
		end
		
		context "Count how many valleys exists" do
			it "return 1" do
				expect(Valleys.count("UDDDUDUU")).to eql(1)
			end
		end
		
		context "Count how many valleys exists" do
			it "return 2" do
				expect(Valleys.count("DDUUDDUDUUUD")).to eql(2)
			end
		end
	end
end
